tests/integration_python/test_edge_cases.py

```diff
@@ -470,6 +470,362 @@ def test_pypi_url_fragment_in_project_deps(tmp_pixi_workspace: Path, pixi: Path)
+
+    # Should fail - invalid hash format
+    verify_cli_command(
+        [pixi, "install", "-v"],
+        cwd=tmp_pixi_workspace,
+        expected_exit_code=ExitCode.FAILURE,
+        stderr_contains=["Hash verification failed", "not a valid hex string"],
+    )
+
+
+def test_pypi_external_source_empty_hash(tmp_pixi_workspace: Path, pixi: Path) -> None:
+    """Test that empty hash value fails validation"""
+    pyproject_content = """
+[project]
+version = "0.1.0"
+name = "test"
+requires-python = "== 3.12"
+dependencies = [
+    "requests @ https://files.pythonhosted.org/packages/f9/9b/335f9764261e915ed497fcdeb11df5dfd6f7bf257d4a6a2a686d80da4d54/requests-2.32.3-py3-none-any.whl#sha256="
+]
+[tool.pixi.workspace]
+platforms = ["linux-64", "osx-arm64", "win-64"]
+channels = ["https://prefix.dev/conda-forge"]
+"""
+    pyproject_path = tmp_pixi_workspace / "pyproject.toml"
+    pyproject_path.write_text(pyproject_content)
+
+    src_dir = tmp_pixi_workspace / "src" / "test"
+    src_dir.mkdir(parents=True, exist_ok=True)
+    (src_dir / "__init__.py").touch()
+
+    # Should fail - empty hash value
+    verify_cli_command(
+        [pixi, "install", "-v"],
+        cwd=tmp_pixi_workspace,
+        expected_exit_code=ExitCode.FAILURE,
+        stderr_contains=["Hash verification failed", "Empty"],
+    )
+
+
+def test_pypi_external_source_unknown_hash_algorithm(tmp_pixi_workspace: Path, pixi: Path) -> None:
+    """Test that unknown hash algorithms fail validation"""
+    pyproject_content = """
+[project]
+version = "0.1.0"
+name = "test"
+requires-python = "== 3.12"
+dependencies = [
+    "requests @ https://files.pythonhosted.org/packages/f9/9b/335f9764261e915ed497fcdeb11df5dfd6f7bf257d4a6a2a686d80da4d54/requests-2.32.3-py3-none-any.whl#sha512=abcdef1234567890"
+]
+[tool.pixi.workspace]
+platforms = ["linux-64", "osx-arm64", "win-64"]
+channels = ["https://prefix.dev/conda-forge"]
+"""
+    pyproject_path = tmp_pixi_workspace / "pyproject.toml"
+    pyproject_path.write_text(pyproject_content)
+
+    src_dir = tmp_pixi_workspace / "src" / "test"
+    src_dir.mkdir(parents=True, exist_ok=True)
+    (src_dir / "__init__.py").touch()
+
+    # Should fail - unknown hash algorithm
+    verify_cli_command(
+        [pixi, "install", "-v"],
+        cwd=tmp_pixi_workspace,
+        expected_exit_code=ExitCode.FAILURE,
+        stderr_contains=["Hash verification failed", "Unsupported hash algorithm"],
+    )
+
+
+def test_pypi_external_source_truncated_hash(tmp_pixi_workspace: Path, pixi: Path) -> None:
+    """Test that truncated hash values fail verification"""
+    pyproject_content = """
+[project]
+version = "0.1.0"
+name = "test"
+requires-python = "== 3.12"
+dependencies = [
+    "requests @ https://files.pythonhosted.org/packages/f9/9b/335f9764261e915ed497fcdeb11df5dfd6f7bf257d4a6a2a686d80da4d54/requests-2.32.3-py3-none-any.whl#sha256=70761cfe03c773ceb22aa2f671b"
+]
+[tool.pixi.workspace]
+platforms = ["linux-64", "osx-arm64", "win-64"]
```

Comment:
To reduce the overhead of testing we could include only the current platform here.